test/test_helper.rb

```diff
@@ -119,6 +119,16 @@ def after_teardown
     Foreman::Plugin.send(:clear, @plugins_backup, @registries_backup)
     @clear_plugins = nil
   end
+
+  def assert_sql_queries(num_of_queries)
+    queries = []
+    ActiveSupport::Notifications.subscribe("sql.active_record") do |name, start, finish, id, payload|
+      queries << payload[:sql] if payload[:sql] =~ /SELECT/
```

Comment:
shouldn't this also catch `UPDATE` and `DROP` and `INSERT` statements?

(but probably out of scope right now)

Reply:
It could be argued that those are SQL "statements" and not "queries" :)

Reply:
Given `SELECT` can change data (hello triggers!), I see no difference. But as said, this is probably more than was asked in the original comment by Ewoud :)